cmd/restic/cmd_key_add.go

```diff
@@ -0,0 +1,127 @@
+package main
+
+import (
+	"context"
+	"os"
+	"strings"
+
+	"github.com/restic/restic/internal/errors"
+	"github.com/restic/restic/internal/repository"
+	"github.com/spf13/cobra"
+)
+
+var cmdKeyAdd = &cobra.Command{
+	Use:   "add",
+	Short: "Add a new key (password) to the repository; returns the added key ID",
+	Long: `
+The "add" sub-command creates a new key and validates the key. Returns the new key ID.
+
+EXIT STATUS
+===========
+
+Exit status is 0 if the command is successful, and non-zero if there was any error.
+	`,
+	DisableAutoGenTag: true,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return runKeyAdd(cmd.Context(), globalOptions, addKeyOpts)
+	},
+}
+
+type NewPasswordOptions struct {
```

Comment:
There is no longer a reason to keep this as a separate struct. Please inline it into `AddKeyOptions`.